freqtrade/enums/tradetype.py

```diff
@@ -0,0 +1,11 @@
+from enum import Enum
+
+
+class TradeType(Enum):
+    """
+    Enum to distinguish between buy, sell, short and exit_short trades
+    """
+    BUY = "buy"
+    SELL = "sell"
+    SHORT = "short"
+    EXIT_SHORT = "exit_short"
```

Comment:
I'd not do this with a enum since there's only 2 relevant values (see below) - but as you've already added it the enum will be fine.

There is however only 2 sides when determining a price - either buy or sell.

It's a short / exit short if you inverse the sequence - but that will not make any difference to how you determine the price at which to buy/sell.

Reply:
I was planning on updating the function to look like this 

```
    def get_rate(self, pair: str, refresh: bool, side: TradeType) -> float:
        """
        Calculates bid/ask target
        bid rate - between current ask price and last price
        ask rate - either using ticker bid or first bid based on orderbook
        or remain static in any other case since it's not updating.
        :param pair: Pair to get rate for
        :param refresh: allow cached data
        :param side: "buy", "sell", "short", or "exit_short"
        :return: float: Price
        :raises PricingError if orderbook price could not be determined.
        """

        if side == TradeType.BUY:
            cache_rate: TTLCache = self._buy_rate_cache
            [strat_name, name, last_balance] = ['bid_strategy', 'Buy', 'ask_last_balance']
        elif side == TradeType.SELL:
            cache_rate: TTLCache = self._sell_rate_cache
            [strat_name, name, last_balance] = ['ask_strategy', 'Sell', 'bid_last_balance']
        elif side == TradeType.SHORT:
            cache_rate: TTLCache = self._short_rate_cache
            [strat_name, name, last_balance] = ['short_strategy', 'Short', 'bid_last_balance']
        elif side == TradeType.EXIT_SHORT:
            cache_rate: TTLCache = self._exit_short_rate_cache
            [strat_name, name, last_balance] = ['exit_short_strategy', 'Exit short', 'ask_last_balance']

        if not refresh:
            rate = cache_rate.get(pair)
            # Check if cache has been invalidated
            if rate:
                logger.debug(f"Using cached {side.value} rate for {pair}.")
                return rate

        conf_strategy = self._config.get(strat_name, {})

        if conf_strategy.get('use_order_book', False) and ('use_order_book' in conf_strategy):

            order_book_top = conf_strategy.get('order_book_top', 1)
            order_book = self.fetch_l2_order_book(pair, order_book_top)
            logger.debug('order_book %s', order_book)
            # top 1 = index 0
            try:
                rate = order_book[f"{conf_strategy['price_side']}s"][order_book_top - 1][0]
            except (IndexError, KeyError) as e:
                logger.warning(
                    f"{name} Price at location {order_book_top} from orderbook could not be "
                    f"determined. Orderbook: {order_book}"
                )
                raise PricingError from e

            logger.info(f"{name} price from orderbook {conf_strategy['price_side'].capitalize()}"
                        f"side - top {order_book_top} order book {side.value} rate {rate:.8f}")
        else:
            logger.info(f"Using Last {conf_strategy['price_side'].capitalize()} / Last Price")
            ticker = self.fetch_ticker(pair)
            ticker_rate = ticker[conf_strategy['price_side']]
            if ticker['last']:
                if side == TradeType.BUY and ticker_rate > ticker['last']:
                    balance = conf_strategy[last_balance]
                    ticker_rate = ticker_rate + balance * (ticker['last'] - ticker_rate)
                elif side == TradeType.SELL and ticker_rate < ticker['last']:
                    balance = conf_strategy.get(last_balance, 0.0)
                    ticker_rate = ticker_rate - balance * (ticker_rate - ticker['last'])
            rate = ticker_rate

        if rate is None:
            raise PricingError(f"{name}-Rate for {pair} was empty.")
        cache_rate[pair] = rate

        return rate
 ```
 
 But are you saying that there should just be two caches? I don't really understand what everything in this function was doing, so I thought it might be better to keep shorts separate from buys, but I'm fine taking it out

Reply:
this function is determining the price.
As such, it's tightly coupled with the orderbook (which in the end, gives you the best buy / sell price).

There's no seperate orderbook for shorts (there's only one orderbook per market)- so i don't see why this method would need this differentiation either.

Think about it this way - either you buy for 100$ - or you sell for 101$. 
If it's a short or long will depend on the sequence of these actions - not on the price.